src/tool.rs

```diff
@@ -199,21 +199,12 @@ impl Tool {
             compiler_detect_output.warnings = compiler_detect_output.debug;
 
             let stdout = run_output(
-                Command::new(path).arg("-E").arg(tmp.path()),
+                Command::new(path).arg("-E").arg("--").arg(tmp.path()),
```

Comment:
I remember that there are some compiler that doesn't work with `--`

I think it's better to detect the clang-cl and use `--` for it?

cc @madsmtm

Reply:
I recall that too, see also discussion in https://github.com/rust-lang/cc-rs/pull/1328.

I feel like we're shooting ourselves in the foot here by not having better CI for this sort of stuff? At the very least, I'd expect a test to be added that doesn't work before the PR, and works after.